test/test_eager_transforms.py

```diff
@@ -1782,6 +1774,44 @@ def composite_output(x):
             assert isinstance(out[i][0], tuple) and \
                 isinstance(out[i][0][1], dict)
 
+    def test_aux_tensor(self, device):
+
+        x = torch.randn(3, device=device)
+        t = torch.randn(3, device=device)
+
+        with self.assertRaisesRegex(
+            RuntimeError, r'jvp\(f, primals, tangents\): output of function f should be a tuple'
+        ):
+            jvp(lambda t: [t, t], (x, ), (t, ), has_aux=True)
+
+        with self.assertRaisesRegex(
+            RuntimeError, r'jvp\(f, primals, tangents\): output of function f should be a tuple'
+        ):
+            jvp(lambda t: (t, t + 2, t + 3), (x, ), (t, ), has_aux=True)
+
+        def f(z):
+            y = z.sin()
+            return y, z.cos()
+
+        out, jvp_out, aux = jvp(f, (x, ), (t, ), has_aux=True)
+        self.assertEqual(aux, x.cos())
+        self.assertEqual(out, x.sin())
+        self.assertEqual(jvp_out, t * x.cos())
+
+    def test_aux_pytree(self, device):
+        def f(x):
+            y = x.sin()
+            return y, {'a': x.cos(), 'b': [x.tan()]}
+
+        x = torch.randn(3, device=device)
+        t = torch.randn(3, device=device)
+
+        out, jvp_out, aux = jvp(f, (x, ), (t, ), has_aux=True)
+        expected_out, expected_aux = f(x)
+        self.assertEqual(out, expected_out)
+        self.assertEqual(aux, expected_aux)
+        self.assertEqual(jvp_out, t * x.cos())
+
```

Comment:
Sorry this is my fault because I didn't write the tests for these, but since we are here... could you also add a test for vjp that tests the error case (that it errors out when aux is not a pytree-of-tensors) and tests for grad with has_aux = True? I'm also happy to handle this as a followup if you don't want to.